src/plan_expand_hypertable.c

```diff
@@ -846,23 +846,59 @@ collect_quals_walker(Node *node, CollectQualCtx *ctx)
 	return expression_tree_walker(node, collect_quals_walker, ctx);
 }
 
-static List *
-find_children_oids(HypertableRestrictInfo *hri, Hypertable *ht, LOCKMODE lockmode)
+static int
+chunk_cmp_chunk_id(const void *c1, const void *c2)
 {
-	/*
-	 * Using the HRI only makes sense if we are not using all the chunks,
-	 * otherwise using the cached inheritance hierarchy is faster.
-	 */
-	if (!ts_hypertable_restrict_info_has_restrictions(hri))
-		return find_inheritance_children(ht->main_table_relid, lockmode);
+	return (*(Chunk **) c1)->fd.id - (*(Chunk **) c2)->fd.id;
+}
+
+static Chunk **
+find_children_chunks(HypertableRestrictInfo *hri, Hypertable *ht, LOCKMODE lockmode,
+					 unsigned int *num_chunks)
+{
+	if (TS_HYPERTABLE_IS_INTERNAL_COMPRESSION_TABLE(ht))
+	{
+		/*
+		 * FIXME somehow chunk lookup doesn't work for internal compression
+		 * tables
+		 */
+		List *chunk_oids = find_inheritance_children(ht->main_table_relid, lockmode);
```

Comment:
Not sure why the normal chunk lookup e.g. `ts_chunk_find_all` doesn't work in this case. I'll have to ask @gayyappan or other compression experts.